### PR TITLE
Add specification for per-axis quantization scheme

### PR DESCRIPTION
**Pull request by sdasgup3**
This PR **partially** solves #1574 in the sense that it just proposes a specification for per-axis quantitation scheme.
In the current proposal  we defined the semantics as part of quantize/dequantize meta functions avoiding text descriptions to provide more clarity. 

Please let me know your feedback.

### CODE REVIEW

**Review by loganchien — APPROVED** [review]

File: docs/spec.md [hunk 1]
```diff
@@ -6317,33 +6317,47 @@ def baseline_type(x: Value | Placeholder | Type) -> Type:
...
-works for per-tensor quantization. Per-axis quantization is work in progress
-([#1574](https://github.com/openxla/stablehlo/issues/1574)).
+associated with the quantized element type.
 
 ```python
+def get_zero_points(quantized_type, result_type):
+  if is_per_tensor_quantized(quantized_type):
+    return broadcast_in_dim(constant(zero_point(quantized_type), storage_type(quantized_type)), [], result_type)
+  if is_per_axis_quantized(quantized_type):
+    for i in index_space(result_type):
```
Comment (rewu93) [inline docs/spec.md:6327]: I guess we need this loop for multi-axis quantized tensors?

Comment (sdasgup3) [reply docs/spec.md:6327]: That is right. FWIW, [here](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#shape-computations ) we define `index_space` which helps us to use a linearized loop over the indices of tensor travered lexicographically.

**Review by rewu93 — APPROVED** [review]

**Review by doyeonkim0 — APPROVED** [review]

File: docs/spec.md [hunk 2]
```diff
@@ -6317,33 +6317,47 @@ def baseline_type(x: Value | Placeholder | Type) -> Type:
...
+
+def get_scales(quantized_type, result_type):
+  if is_per_tensor_quantized(quantized_type):
+    return broadcast_in_dim(constant(scale(quantized_type), expressed_type(quantized_type)), [],
+            type(result_type))
+  if is_per_axis_quantized(quantized_type):
+    for i in index_space(result_type):
+      d = quantized_dimension(quantized_type)
+      scales[i] = scales(quantized_type)[i[d]]
+    return scales
```
[the remark this answers is not in the document]
Comment (sdasgup3) [reply docs/spec.md:6340]: You are right!
`for i in index_space(result_type):` returns `(0,0), (0,1), (0,2), (1,0), (1,1), (1,2)` and for each such index `i` we figure out the scale value `scales(quantized_type)[i[d]]` and populate the scale tensor of shape `[2,3]`.

[the remark this answers is not in the document]
Comment (dansuh17) [reply docs/spec.md:6340]: Thanks for clarifying!

**Review by dansuh17 — APPROVED** [review]

File: docs/spec.md [hunk 3]
```diff
@@ -6317,33 +6317,47 @@ def baseline_type(x: Value | Placeholder | Type) -> Type:
 floating-point tensor types. This happens via converting quantized elements
 which represent integer values of the storage type into corresponding
 floating-point values of the expressed type using the zero point and scale
-associated with the quantized element type. At the moment, this function only
-works for per-tensor quantization. Per-axis quantization is work in progress
-([#1574](https://github.com/openxla/stablehlo/issues/1574)).
+associated with the quantized element type.
 
 ```python
+def get_zero_points(quantized_type, result_type):
```
Comment (burmako) [inline docs/spec.md:6323]: Let's rename to `compute_zero_points` and `compute_scales`? `get_` suggests that this is a getter, whereas there's a bit more going on than just getting.

**Review by burmako — APPROVED** [review]

File: docs/spec.md (hunk 3 above)
Comment (sdasgup3) [reply docs/spec.md]: done. Thanks!

**Pull request merged**